src/configuration.ts

```diff
@@ -650,16 +678,26 @@ let configurationBuildLog: string | undefined;
 export function getConfigurationBuildLog(): string | undefined { return configurationBuildLog; }
 export function setConfigurationBuildLog(name: string): void { configurationBuildLog = name; }
 
+let configurationPreConfigureArgs: string[] = [];
+export function getConfigurationPreConfigureArgs(): string[] { return configurationPreConfigureArgs; }
+export function setConfigurationPreConfigureArgs(args: string[]): void { configurationPreConfigureArgs = args; }
+
+let configurationPostConfigureArgs: string[] = [];
+export function getConfigurationPostConfigureArgs(): string[] { return configurationPostConfigureArgs; }
+export function setConfigurationPostConfigureArgs(args: string[]): void { configurationPostConfigureArgs = args; }
+
 // Analyze the settings of the current makefile configuration and the global workspace settings,
 // according to various merging rules and decide what make command and build log
 // apply to the current makefile configuration.
 async function analyzeConfigureParams(): Promise<void> {
     getBuildLogForConfiguration(currentMakefileConfiguration);
     await getCommandForConfiguration(currentMakefileConfiguration);
     getProblemMatchersForConfiguration(currentMakefileConfiguration);
+    getPreConfigureArgsForConfiguration(currentMakefileConfiguration);
```

Comment:
Didn't test and debug yet, will understand better soon. But why have the getters of pre/post config args here? analyzeConfigParams are about configure phase (which invokes make --dry-run plus other switches which get calculated in here) and not about pre or post configure which doesn't even call "make --dry-run" but whatever is in the pre/post configure script. Eventually, maybe we need something like "analyzePrePostConfigureParams".

Reply:
This was done to ensure that the preConfigure and postConfigure settings are correctly established. I viewed these as directly related to configure, since it's directly tied to before configure and after configure.